Try to make README more approachable. (#235)
dart-lang/jnigen: README.md

The configuration for binding generation is usually provided through YAML. A programmatic API (`package:jnigen`) is also provided.

Three configuration details are needed to generate the bindings. Everything else is optional:

* _Inputs_: input can be Java source files (`source_path`), or compiled classes / JARs (`class_path`). Some maven / gradle based tooling is also provided to simplify obtaining dependencies.

* _Outputs_: Output can be generated in package-structured (one file per class) or single file bindings. Target path to write C and Dart bindings needs to be specified.

* _Classes_: Specify which classes or packages you need bindings for. Specifying a package includes all classes inside it recursively.

Check out the [examples](jnigen/example/) to see some sample configurations.

C code is always generated into a directory with it's own build configuration. It's built as a separate dynamic library.

> Lastly, [dart_only bindings](#pure-dart-bindings) mode is also available as a proof-of-concept, which does not need intermediate C bindings apart from dependency on the support library `package:jni`.

Reviewer: ```suggestion
Lastly, [dart_only bindings](#pure-dart-bindings) mode is also available as a proof-of-concept. It does not need intermediate C bindings, only a dependency on the support library `package:jni`.
```

## Example

> It's possible to generate bindings for libraries, or Java source files.

Reviewer: libraries? Did you mean JAR files?

Lastly, some libraries such as `java.awt` do not exist in android. Attempting to use libraries which depend on them can also lead to ClassNotFound errors.

#### `jnigen` is not finding classes.
Ensure you are providing correct source and class paths, and they follow standard directory structure. If your class name is `com.abc.MyClass`, `MyClass` must be in `com/abc/MyClass.java` relative to one of the source paths, or `com/abc/MyClass.class` relative to one of the class paths specified in YAML.

If the classes are in JAR file, make sure to provide path to JAR file itself, and not the containing directory.

#### `jnigen` is unable to parse sources.
If the errors are similar to `symbol not found`, ensure all dependencies of the source are available. If such dependency is compiled, it can be included in `class_path`.

#### Should I use `jnigen` over Method channels?

> This is currently an experimental package. Many features are missing, and it's rough around the edges. You're welcome to try it and give feedback.

Reviewer: @HosseinYousefi What is the preferred wording for where we are with this product?

Reviewer 2: This looks good to me. @mit-mit What do you think?

The JAR files (`$SDK_ROOT/platforms/android-$VERSION/android.jar`) can be used instead. But compiled JARs do not include JavaDoc and method parameter names. This JAR is automatically included by Gradle when `android_sdk_config` >> `add_gradle_deps` is specified.

## Contributing

> See the wiki for architecture-related documents.

Reviewer: We should add a CONTRIBUTING.md (can do in a separate PR.)

For inspiration: https://github.com/dart-lang/ffigen/blob/master/CONTRIBUTING.md

The markdown file can link to the wiki.
